nf_core/modules/module_test.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python
+"""
+The ModulesTest class runs the tests locally
+"""
+
+import logging
+import questionary
+import os
+import pytest
+import sys
+import rich
+import subprocess
+import shlex
+from pathlib import Path
+
+import nf_core.utils
+from .modules_repo import ModulesRepo
+
+log = logging.getLogger(__name__)
+
+
+class ModulesTest(object):
+    """
+    Class to run module pytests.
+
+    ...
+
+    Attributes
+    ----------
+    module_name : str
+        name of the tool to run tests for
+    no_prompts : bool
+        flat indicating if prompts are used
+    pytest_args : tuple
+        additional arguments passed to pytest command
+
+    Methods
+    -------
+    run():
+        Run test steps
+    _check_inputs():
+        Check inputs. Ask for module_name if not provided and check that the directory exists
+    _set_profile():
+        Set software profile
+    _run_pytests(self):
+        Run pytest
+    """
+
+    def __init__(
+        self,
+        module_name=None,
+        no_prompts=False,
+        pytest_args="",
+    ):
+        self.module_name = module_name
+        self.no_prompts = no_prompts
+        self.pytest_args = pytest_args
+
+    def run(self):
+        """Run test steps"""
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self._check_inputs()
+        self._set_profile()
+        self._check_profile()
+        self._run_pytests()
+
+    def _check_inputs(self):
+        """Do more complex checks about supplied flags."""
+
+        # Get the tool name if not specified
+        if self.module_name is None:
+            if self.no_prompts:
+                raise UserWarning(
+                    "Tool name not provided and prompts deactivated. Please provide the tool name as TOOL/SUBTOOL or TOOL."
+                )
+            modules_repo = ModulesRepo()
+            modules_repo.get_modules_file_tree()
+            self.module_name = questionary.autocomplete(
+                "Tool name:",
+                choices=modules_repo.modules_avail_module_names,
+                style=nf_core.utils.nfcore_question_style,
+            ).ask()
+        module_dir = Path("modules") / self.module_name
+
+        # First, sanity check that the module directory exists
+        if not module_dir.is_dir():
+            raise UserWarning(f"Cannot find directory '{module_dir}'. Should be TOOL/SUBTOOL or TOOL")
```

Comment:
```suggestion
            raise UserWarning(f"Cannot find directory '{module_dir}'. Should be TOOL/SUBTOOL or TOOL. Are you running the tests inside the nf-core/modules main directory?")
```

Reply:
it has happened to me a couple of times that I wasn't in the nf-core/modules directory so maybe a message like this would help